### ISSUE TITLE
ROCm: resolve malformed rocprofv3 argument export before trace attribution

### ISSUE THREAD

**Issue by VikashLoomba** [mudler/vllm.cpp#3051]
Row: `BACKEND-ROCM`

The #3041 matched-profiler prerequisite exposed a malformed export from the approved oracle image's rocprofv3 1.3.2. The native baseline compiled inside that unchanged image and passed its GPU backend suite both unprofiled and profiled: 11 cases, 1,065 assertions in each process. The profiler exited 0 and finalized its output files. This does not make every exported field valid evidence.

The 26,561,668-byte `native-backend_results.json` is not valid UTF-8. Decoding fails at byte 25,561,946 inside the `kname` value for `hipModuleGetFunction`, correlation ID 28989. The corresponding CSV records that API call. The raw JSON SHA256 is `b8bc62a074369d37f6aefdeeddbc21fbe75ee55af0bfe3d176e872dab54744ac`. No replacement decoding or rewritten artifact was used.

Reproduction bindings: native source `c94509c87e35ae80b24419687d2a3106ec923f6d`; library SHA256 `1dd2deb5f861610088ad5d63c2ce77ca390cc7f39f3f33e0adf5360579ca30b3`; test binary SHA256 `f38f157ed230843d54fbd27b2e933c6f4279d66c85c5df6ea5d1a331787de9c8`; image `sha256:80aab4c182a1f3eeebe286173977e57fcaf10a049b41f475655b35d285de31dc`. Build: GCC 13.4 and HIP Clang 22, gfx1100, Release, jobs 4. Run at 2026-09-08 00:08 UTC under the local GPU mutex with the existing image environment:

```text
/opt/rocm/bin/rocprofv3 --sys-trace --output-config 1 --output-format json csv --output-directory <owned-output> --output-file native-backend -- <same-image-built-test_rocm_backend>
```

The UTF-8 CSV files parse and contain 24,049 HIP API records, including five graph launches, and 19 kernel-dispatch records. They establish a narrower trace surface. They do not repair or validate the malformed JSON argument field, establish its cause, or prove complete invocation operands. The fixture loads no model and makes no throughput or cross-engine parity claim.

#3041 owns resolving the export/argument-attribution prerequisite before relying on those values. Preserve the original malformed bytes, diagnose the exact exporter/argument lifetime, and select a reviewed capture route that supplies the required fields without altering the oracle or weakening acceptance. A cache file or kernel name alone cannot substitute for loaded-code/operand evidence. The #2773 causal-convolution investigation retains the same requirement for its own executing path. No SDK patch, runtime replacement, or decoding tolerance is implied by this issue.

**Comment by VikashLoomba** [comment]
A second native reproducer confirms the malformed export on the model path. The unchanged approved image ran candidate `c3c880d6abdfc6c698cee7e3bb1bbf4f73369062` with Qwen3.5-0.8B, auto KV, and graph capture. The application and profiler exited 0. Captured state, pre-sampling logits, tokens, and metadata match the independent unprofiled run exactly.

The 53,196,322-byte `native_results.json` fails strict UTF-8 decoding at byte 27,943,435. SHA256: `b81024a7b609aea2acf229bee08c7f5ba9bafb216d495dd4bbce70e2de00e502`. The invalid byte is `0x81` inside `hipModuleGetFunction`'s `kname` argument. The entry records return value 500 and correlation ID 28188. The corresponding CSV uniquely identifies that call on process and thread 19. The original bytes remain unchanged.

The UTF-8 CSV exports parse: 41,425 HIP API rows, 93,988 HSA API rows, 6,477 kernel rows, 733 allocation rows, and 456 copy rows. Six `hipGraphLaunch` calls each join 384 kernel rows by correlation ID and thread. These are partial CSV observations. They do not establish complete argument attribution, invocation parity, or performance acceptance. The attempted post-run loaded-library map missed the completed process and supplies no map evidence.

The command retains the approved wrapper: `rocprofv3 --sys-trace --output-config 1 --output-format json csv`, followed by the ordinary native capture driver. Image: `sha256:80aab4c182a1f3eeebe286173977e57fcaf10a049b41f475655b35d285de31dc`. Native binary SHA256: `9dfcd1b3f37dde4cd0ffd2e5f5af6ff61f00c2472ad5ff87fba06990f9f7272c`. Exact state comparison report SHA256: `b00e21fbceac9b10d14f2beba38de30a5fc7156626d2424fac9bfbebd41e6f6b`.

Full trace validation remains failing under this exporter. The required trace gate remains open. No replacement decoding, exporter patch, oracle change, or relaxed acceptance was used.

**Comment by VikashLoomba** [comment]
A read-only source audit found a possible argument-lifetime cause. The approved image reports SDK revision `ROCm/rocm-systems@2b22ab0195cc1461cd9abf3b969e9dd7c10af350`.

At that revision, [the HIP wrapper](https://github.com/ROCm/rocm-systems/blob/2b22ab0195cc1461cd9abf3b969e9dd7c10af350/projects/rocprofiler-sdk/source/lib/rocprofiler-sdk/hip/hip.cpp#L318) copies the original argument struct into an extended buffered record. [The buffer](https://github.com/ROCm/rocm-systems/blob/2b22ab0195cc1461cd9abf3b969e9dd7c10af350/projects/rocprofiler-sdk/source/lib/common/container/record_header_buffer.hpp#L309) copy-constructs that record. The tool retains the copied record until output.

[The serializer](https://github.com/ROCm/rocm-systems/blob/2b22ab0195cc1461cd9abf3b969e9dd7c10af350/projects/rocprofiler-sdk/source/include/rocprofiler-sdk/cxx/serialization/save.hpp#L610) obtains argument strings when it saves the record. The buffered HIP iterator passes max_deref=0, but [the common stringizer](https://github.com/ROCm/rocm-systems/blob/2b22ab0195cc1461cd9abf3b969e9dd7c10af350/projects/rocprofiler-sdk/source/lib/common/stringize_arg.hpp#L59) handles C strings before its bounded pointer branches and reads the pointed-to bytes. A borrowed kname that changes or expires before output is therefore a concrete source hypothesis.

This does not yet attribute the physical invalid byte to that lifetime mechanism or prove the source revision matches every loaded binary. The original process-map attempt missed the exited process. The next proof needs the actual SDK buffer/serializer with controlled string lifetime, followed by a live argument/loaded-library binding. The source audit binds 22 retrieved files by Git blob identity and SHA256; no profiler, runtime, decode path, or trace gate was changed.
